### PR TITLE
make it possible to create a FITS header in memory

### PR DESCRIPTION
**Pull request by gervandiepen**
Solving issue #270

### CODE REVIEW

**Comment by tammojan** [comment]
@gervandiepen , can you assign a volunteer to review this?

**Comment by dpetry** [comment]
Thanks. I was already looking at it but wanted to wait until all checks had finished.
And yesterday I was not at work.

File: images/Images/ImageFITS2Converter.cc [hunk 1]
```diff
@@ -804,13 +808,13 @@ Bool ImageFITSConverter::ImageToFITSOut(
 	// and make Stokes the very last if requested
 	//
 	IPosition shape = image.shape();
-	IPosition newShape = shape;
+        fhi.newShape = shape;
```
Comment (tammojan) [inline images/Images/ImageFITS2Converter.cc]: Should have been indented with tabs here, like the rest of the file. Sorry for nitpicking..

File: images/Images/ImageFITS2Converter.cc [hunk 2]
```diff
@@ -842,52 +846,51 @@ Bool ImageFITSConverter::ImageToFITSOut(
...
+				fhi.needNonOptimalCursor=True;
 			}
 		}
 
 	}
 	//
-	Bool applyMask = False;
-	Array<Bool>* pMask = 0;
+	fhi.applyMask = False;
+        fhi.pMask = 0;
```
Comment (tammojan) [inline images/Images/ImageFITS2Converter.cc]: Same here, should have been tabs.

**Comment by dpetry** [comment]
Ger, so you introduced the new struct ImageFITSHeaderInfo in roder to encapsulate all information
pertaining to the header such that this can be kept in memory.
This led to several interface changes.
Have you tried to build CASA with this?
If not, I should probably try that before we merge.

**Comment by gervandiepen** [comment]
The original ImageToFITS function has been split into 2 parts with the
ImageFITSHeaderInfo struct as the link between the two. Thus the original
API has not changed.
But it is certainly a good idea to build CASA before doing the merge.

I'll take a look at the indentation.

**Comment by juliantaylor** [comment]
casa builds and tests fine with this change

**Comment by dpetry** [comment]
You also mean the NRAO CASA including code, asap, and gcwrap, right?

**Comment by juliantaylor** [comment]
yes

**Comment by dpetry** [comment]
Then I think, pending the things Tammo found, this is good to go.

**Comment by dpetry** [comment]
Ger, let me know when you think the cosmetic parts are OK.

**Comment by gervandiepen** [comment]
Re-indented all of ImageFITS2Converter.tcc to get it right.

**Pull request merged**